app/components/search_result/document_gallery_component.rb

```diff
@@ -9,5 +9,27 @@ def preview_container_dom_class
     def classes
       super - ['document'] + ['gallery-document']
     end
+
+    def data
+      {
+        doc_id: @document.id,
+        document_counter: @counter,
+        preview_in_gallery: 'true'
```

Comment:
```suggestion
        document_counter: @counter
```

I think you determined this property is ignored in #4986?

Reply:
Yes. This falls out after I rebased.